### PR TITLE
Fix the issue with null coalescing fields as input parameters.

### PR DESCRIPTION
**Pull request by lfu**
https://bugzilla.redhat.com/show_bug.cgi?id=1698184

@miq-bot add_label bug, hammer/no, changelog/yes

### CODE REVIEW

**Comment by gmcculloug** [comment]
@lfu Can you correct the `Do not use prefix _` robocop issues.  The other item was there before.

**Comment by gmcculloug** [comment]
bump @mkanoor @tinaafitz Any thoughts on this one?

**Comment by tinaafitz** [comment]
@mkanoor Please review.

File: lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ def self.invoke(obj, aem, args)
       aem.inputs.each do |f|
         key   = f.name
         value = args[key]
-        value = obj.attributes[key] || obj.substitute_value(f.default_value) if value.nil?
+        value = obj.attributes[key] || obj.get_value(f) if value.nil?
```
Comment (mkanoor) [inline lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb]: @lfu does this field type have substitute set to true or nil?

Comment (lfu) [reply lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb]: #<MiqAeField id: 48440, aetype: nil, name: "arg2", display_name: nil, datatype: "null coalescing", priority: 2, owner: nil, default_value: "${#var3} || ${#var2} || ${#var1} ", substitute: false, message: "create", visibility: nil, created_on: "2019-06-27 18:01:41", updated_on: "2019-08-20 19:22:05", class_id: nil, collect: nil, method_id: 11574, scope: nil, description: nil, condition: nil, on_entry: nil, on_exit: nil, on_error: nil, max_retries: nil, max_time: nil, updated_by: "admin", updated_by_user_id: 1>

Comment (lfu) [reply lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb]: @mkanoor for null coalescing data type, substitute is false.

Comment (mkanoor) [reply lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb]: @lfu 👍 
The NULL coalescing calls substitution directly

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit abc51bc36a668bd672eb061033b84d070011d122
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Wed Aug 21 12:01:25 2019 -0400

    Merge pull request #334 from lfu/null_coalescing_1698184
    
    Fix the issue with null coalescing fields as input parameters.
    
    (cherry picked from commit 69b58ee4ce23eef1591a90c145e8b225821a80b9)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1698184
```

**Pull request merged**